sympy/core/numbers.py

```diff
@@ -1254,7 +1254,7 @@ def _eval_is_infinite(self):
         return False
 
     def _eval_is_integer(self):
-        return self._mpf_ == fzero
+        return self._mpf_[2] == fzero[2]
```

Comment:
This will only work with odd integers. For all integers, we have `self._mpf_[2] >= 0`.